From d798915f05b3e6c72d34aea269ff1e36eb2b57bb Mon Sep 17 00:00:00 2001
From: Eric Lippmann <eric.lippmann@icinga.com>
Date: Tue, 3 Sep 2019 10:55:19 +0200
Subject: [PATCH 1/3] Add yet empty class Str

This should be the place for various string manipulation functions.
---
 src/Str.php | 10 ++++++++++
 1 file changed, 10 insertions(+)
 create mode 100644 src/Str.php

diff --git a/src/Str.php b/src/Str.php
new file mode 100644
index 0000000..ac44366
--- /dev/null
+++ b/src/Str.php
@@ -0,0 +1,10 @@
+<?php
+
+namespace ipl\Stdlib;
+
+/**
+ * Collection of string manipulation functions
+ */
+class Str
+{
+}
-- 
2.22.0


From 1eeccc20049f9e52ad4b451324ae666c70dd5d6d Mon Sep 17 00:00:00 2001
From: Eric Lippmann <eric.lippmann@icinga.com>
Date: Tue, 3 Sep 2019 10:58:37 +0200
Subject: [PATCH 2/3] Add Str::camel()

Converts a string to camel case.
---
 src/Str.php | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/Str.php b/src/Str.php
index ac44366..a218ca5 100644
--- a/src/Str.php
+++ b/src/Str.php
@@ -7,4 +7,19 @@ namespace ipl\Stdlib;
  */
 class Str
 {
+    /**
+     * Convert the given string to camel case
+     *
+     * The given string may be delimited by the following characters: '_' (underscore), '-' (dash), ' ' (space).
+     *
+     * @param string $subject
+     *
+     * @return string
+     */
+    public static function camel($subject)
+    {
+        $normalized = str_replace(['-', '_'], ' ', $subject);
+
+        return lcfirst(str_replace(' ', '', ucwords(strtolower($normalized))));
+    }
 }
-- 
2.22.0


From a53462c622615719a0718766552c735541b3cc06 Mon Sep 17 00:00:00 2001
From: Eric Lippmann <eric.lippmann@icinga.com>
Date: Tue, 3 Sep 2019 11:07:59 +0200
Subject: [PATCH 3/3] Test Str::camel()

---
 tests/php/StrTest.php | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)
 create mode 100644 tests/php/StrTest.php

diff --git a/tests/php/StrTest.php b/tests/php/StrTest.php
new file mode 100644
index 0000000..4a34e8e
--- /dev/null
+++ b/tests/php/StrTest.php
@@ -0,0 +1,29 @@
+<?php
+
+namespace ipl\Tests\Stdlib;
+
+use ipl\Stdlib\Str;
+
+class StrTest extends TestCase
+{
+    public function testCamelDoesNothingIfStringHasNoDelimitersAndIsLowerCase()
+    {
+        $this->assertSame('noop', Str::camel('noop'));
+    }
+
+    public function testCamelFromSnakeCaseString()
+    {
+        $this->assertSame('snakeCase', Str::camel('snake_case'));
+    }
+
+    public function testCamelFromKebabCaseString()
+    {
+
+        $this->assertSame('kebabCase', Str::camel('kebab-case'));
+    }
+
+    public function testCamelFromSpaceDelimitedString()
+    {
+        $this->assertSame('spaceDelimited', Str::camel('space delimited'));
+    }
+}
-- 
2.22.0

